revscoring/scoring/models/model.py

```diff
@@ -0,0 +1,255 @@
+        self.version = version
+
+        self.info = ModelInfo()
+        self.info['type'] = self.__class__.__name__
+        self.info['version'] = version
+        self.info['params'] = self.params
+        self.info['environment'] = environment or Environment()
+        if statistics is not None:
+            self.info['statistics'] = statistics
+
+    def score(self, feature_values):
+        """
+        Make a prediction or otherwise use the model to generate a score.
+
+        :Parameters:
+            feature_values : collection(`mixed`)
+                an ordered collection of values that correspond to the
+                `Feature` s provided to the constructor
+
+        :Returns:
+            A `dict` of statistics
+        """
+        raise NotImplementedError()
+
+    def test(self, values_labels):
+        """
+        Tests the model against a labeled data.
+
+        :Parameters:
+            values_labels : `iterable` (( `<feature_values>`, `<label>` ))
+                an iterable of labeled data Where <values_labels> is an ordered
+                collection of predictive values that correspond to the
+                `Feature` s provided to the constructor
+
+        :Returns:
+            A dictionary of test results.
+        """
+        # Score all of the observations
+        score_labels = [(self.score(values), label)
+                        for values, label in values_labels]
+
+        # Fit builtin statistics engine
+        self.info['statistics'].fit(score_labels)
+
+        return self.info['statistics']
+
+    @classmethod
+    def load(cls, f, error_on_env_check=False):
+        """
+        Reads serialized model information from a file.
+        """
+        if hasattr(f, 'buffer'):
+            model = pickle.load(f.buffer)
+        else:
+            model = pickle.load(f)
+
+        model.info['environment'].check(raise_exception=error_on_env_check)
+        return model
+
+    def dump(self, f):
+        """
+        Writes serialized model information to a file.
+        """
+
+        if hasattr(f, 'buffer'):
+            return pickle.dump(self, f.buffer)
+        else:
+            return pickle.dump(self, f)
+
+    @classmethod
+    def from_config(cls, config, name, section_key='scorer_models'):
+        section = config[section_key][name]
+
+        if 'module' in section:
+            return yamlconf.import_module(section['module'])
+        elif 'class' in section:
+            class_path = section['class']
+            Class = yamlconf.import_module(class_path)
+            if 'model_file' in section:
+                return Class.load(open(section['model_file'], 'rb'))
+            else:
+                return Class(**{k: v for k, v in section.items()
+                                if k != "class"})
+
+
+class Learned(Model):
+
+    def __init__(self, *args, scale=False, center=False, **kwargs):
+        """
+        A machine learned model.  Beyond :class:`revscoring.Model`, this
```

Comment:
Good Q.  Sometimes models are hard-coded.  Not all models are fitted to data. :)  `x * 2 = y` is a model.  You don't need to fit the 2 at all.